frontend/src/container/GridGraphLayout/Graph/FullView/GraphManager.tsx

```diff
@@ -0,0 +1,326 @@
+import { Button, Checkbox, ConfigProvider, Input } from 'antd';
+import { CheckboxChangeEvent } from 'antd/es/checkbox';
+import { ColumnType } from 'antd/es/table';
+import { ChartData, ChartDataset } from 'chart.js';
+import { ResizeTable } from 'components/ResizeTable';
+import { useNotifications } from 'hooks/useNotifications';
+import isEqual from 'lodash-es/isEqual';
+import { memo, useEffect, useState } from 'react';
+
+import {
+	FilterTableAndSaveContainer,
+	FilterTableContainer,
+	SaveContainer,
+} from './styles';
+
+function GraphManager({
+	data,
+	graphVisibilityHandler,
+	name,
+}: GraphManagerProps): JSX.Element {
+	const [graphVisibilityArray, setGraphVisibilityArray] = useState<boolean[]>(
+		Array(data.datasets.length).fill(true),
+	);
+	const [tableDataSet, setTableDataSet] = useState<ExtendedChartDataset[]>(
+		data.datasets.map(
+			(item: ChartDataset) =>
+				({
+					...item,
+					show: true,
+					sum: parseFloat(
+						(item.data as number[]).reduce((a, b) => a + b, 0).toFixed(0),
+					),
+					avg: parseFloat(
+						(
+							(item.data as number[]).reduce((a, b) => a + b, 0) / item.data.length
+						).toFixed(0),
+					),
+					max: parseFloat(Math.max(...(item.data as number[])).toFixed(0)),
+					min: parseFloat(Math.min(...(item.data as number[])).toFixed(0)),
+				} as ExtendedChartDataset),
+		),
+	);
+	const [legendEntries, setLegendEntries] = useState<LegendEntryProps[]>([]);
+
+	const { notifications } = useNotifications();
+
+	const showAllDataSet = (data: ChartData): LegendEntryProps[] =>
+		data.datasets.map(
+			(item) =>
+				({
+					label: item.label,
+					show: true,
+				} as LegendEntryProps),
+		);
+
+	useEffect(() => {
+		if (graphVisibilityHandler) {
+			graphVisibilityHandler([...graphVisibilityArray]);
+		}
+	}, [graphVisibilityArray, graphVisibilityHandler]);
+
+	useEffect(() => {
+		const graphDisplayStatusArray: boolean[] = Array(data.datasets.length).fill(
+			true,
+		);
+		setTableDataSet(
+			data.datasets.map((item: ChartDataset) => ({
+				...item,
+				show: true,
+				sum: Math.floor((item.data as number[]).reduce((a, b) => a + b, 0)),
+				avg: Math.floor(
+					(item.data as number[]).reduce((a, b) => a + b, 0) / item.data.length,
+				),
+				max: Math.floor(Math.max(...(item.data as number[]))),
+				min: Math.floor(Math.min(...(item.data as number[]))),
+			})),
+		);
+		data.datasets.forEach((d, i) => {
+			const index = legendEntries.findIndex((di) => di.label === d.label);
+			if (index !== -1) {
+				graphDisplayStatusArray[i] = legendEntries[index].show;
+			}
+		});
+		setGraphVisibilityArray(graphDisplayStatusArray);
+	}, [data, legendEntries]);
+
+	useEffect(() => {
```

Comment:
what's this code trying to achieve?

Reply:
It is checking whether data (i.e. to enable or disable the legend) for the particular graph is present in the local storage, if it is present then set the graph visibility according to the local storage data in the chart. This should be checked every time the `data` and `name` of the graph changes. `data` will change if we re-fetch the data and `name` will change when we open any other chart.